### PR TITLE
Feature/go go consumer

### PR DESCRIPTION
**Pull request by calumbrodie**

### CODE REVIEW

**Comment by calumbrodie** [comment]
These are the Go Consumers in my local VM
1__vagrant_hunter-dev__vagrant__vagrant_

These are the PHP consumers on staging
1__calum_brodie_hunter-staging____ssh_

**Comment by calumbrodie** [comment]
This branch fails because of the 'realpath' calls which are returning null as the paths in the test don't exist. They do exist in the VM though.... I guess I might need to change the code to skip all of these tests (during tests) :/

**Comment by calumbrodie** [comment]
Tests will pass after merge. If you want to review @shieldo or @gsdevme or anyone else I'm going to merge tomorrow and tag bundle at a new major version

File: Command/ConsumerCommand.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        } else {
+            $message = new AMQPMessage($jsonData);
+        }
+
+        $strict = $input->getOption('strict-exit-code');
+        $consumerReturn = $this->getContainer()->get('markup_job_queue.consumer')->execute($message);
+
+        // if not running in strict mode - always acknowledge the message otherwise it will requeue forever
+        if (!$strict) {
+            exit(self::STRICT_CODE_ACK);
```
Comment (shieldo) [inline Command/ConsumerCommand.php]: A console command `execute()` method should return the exit code, not `exit()` with it (for example, any terminate event won't fire etc etc).

File: Job/ConsoleCommandJob.php [hunk 2]
```diff
@@ -40,20 +48,38 @@ public function run(ContainerInterface $container)
...
                 $message = sprintf('A job `%s` failed with topic `%s` with output:%s and the error output: %s', $command, $this->topic, $process->getOutput(), $process->getErrorOutput());
-                throw new JobFailedException($message);
+                throw new JobFailedException($message, $process->getExitCode());
             }
+            return $process->getOutput();
+        } catch (JobFailedException $e) {
+            throw $e;
         } catch (\Exception $e) {
-            throw new JobFailedException($e->getMessage());
+            throw new JobFailedException($e->getMessage(), 'lol');
```
Comment (shieldo) [inline Job/ConsoleCommandJob.php]: Is this intentional?

File: Reader/QueueReader.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $collection = new ArrayCollection();
+        try{
+            $apiResponse = $this->api->queues()->all($this->vhost);
+            foreach($apiResponse as $q) {
+                $collection->add(Queue::constructFromApiResponse($q));
+            }
+        } catch(\Exception $e) {
+            throw $e;
+            return $collection;
```
Comment (shieldo) [inline Reader/QueueReader.php]: This `return` is unreachable

File: Reader/QueueReader.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    /**
+     * Can a connection to rabbitMq be established at all?
+     *
+     * @return boolean
+     */
+    public function isAlive()
+    {
+        try{
```
Comment (shieldo) [inline Reader/QueueReader.php]: CS ;)

File: Reader/QueueReader.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            $this->api->alivenessTest('test');
+            return true;
+        } catch(\Exception $e) {
```
Comment (shieldo) [inline Reader/QueueReader.php]: here too :(

File: Reader/RecurringConsoleCommandConfigurationJobLogReader.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+     * Get an array of JobLogCollections keyed by configuration uuid
+     *
+     * @param int $maxQuantity
+     * @return array
+     */
+    public function getJobLogCollections($maxQuantity = 5)
+    {
+        $collection = [];
+        $configurations = $this->recurringConsoleCommandReader->getConfigurations();
+        foreach($configurations as $configuration)
```
Comment (shieldo) [inline Reader/RecurringConsoleCommandConfigurationJobLogReader.php]: CS

File: Resources/config/services.yml [hunk 7]
```diff
@@ -14,16 +17,18 @@ services:
             - '@markup_job_queue.scheduled'
     jobby:
         alias: markup_job_queue.manager
-    markup_admin_job_queue_recurring_console_command_reader:
+    markup_job_queue.reader.recurring_console_command:
```
Comment (shieldo) [inline Resources/config/services.yml]: I think calling it `markup_job_queue.recurring_console_command_reader` would be far clearer.

Comment (gsdevme) [inline Resources/config/services.yml]: Shouldn't it actually start with `markup.` ? its in the markup namespace?

**Pull request merged**